runtime/src/jvmTest/kotlin/pbandk/MessageExtensionsJvmTest.kt

```diff
@@ -1,44 +1,99 @@
-        val foo = Foo.getDefaultInstance()
+        val foo = pbandk.testpb.Foo.defaultInstance
         val output = ByteArrayOutputStream(100 * 1000)
 
-        orgMessage.writeDelimitedTo(output)
-        foo.writeDelimitedTo(output)
-        orgMessage.writeDelimitedTo(output)
-        foo.writeDelimitedTo(output)
+        orgMessage.encodeDelimitedToStream(output)
+        foo.encodeDelimitedToStream(output)
+        orgMessage.encodeDelimitedToStream(output)
+        foo.encodeDelimitedToStream(output)
 
         val result = output.toByteArray()
         val input = ByteArrayInputStream(result)
 
-        val m1 = MessageWithMap.parseDelimitedFrom(input)
+        val m1 = MessageWithMap.decodeDelimitedFromStream(input)
         assertEquals(m1, orgMessage)
 
-        val m2 = Foo.parseDelimitedFrom(input)
+        val m2 = pbandk.testpb.Foo.decodeDelimitedFromStream(input)
         assertEquals(m2, foo)
 
-        val m3 = MessageWithMap.parseDelimitedFrom(input)
+        val m3 = MessageWithMap.decodeDelimitedFromStream(input)
         assertEquals(m3, orgMessage)
 
-        val m4 = Foo.parseDelimitedFrom(input)
+        val m4 = pbandk.testpb.Foo.decodeDelimitedFromStream(input)
         assertEquals(m4, foo)
 
-        assertNull(MessageWithMap.parseDelimitedFrom(input))
-        assertNull(Foo.parseDelimitedFrom(input))
+        assertNull(MessageWithMap.decodeDelimitedFromStream(input))
+        assertNull(pbandk.testpb.Foo.decodeDelimitedFromStream(input))
+    }
+
+    @Test
+    fun testReadWithJavaProtoBuf() {
+        val testMap = mapOf("1" to "a", "2" to "b", "blahblahblah" to "5000")
+        val pbMap = MessageWithMap(testMap)
+        val output = ByteArrayOutputStream(100 * 1000)
+
+        // Write pbandk `MessageWithMap` to an output stream
+        pbMap.encodeDelimitedToStream(output)
+
+        // Read protobuf-java `MessageWithMap` instance from the same stream
+        val result = output.toByteArray()
+        val input = ByteArrayInputStream(result)
+        val javaMap = pbandk.testpb.java.Test.MessageWithMap.parseDelimitedFrom(input)
+
+        // Ensure the two instances are equal to each other
+        kotlinJavaRoundtripTest(javaMap, pbMap, pbandk.testpb.MessageWithMap)
+    }
+
+    @Test
+    fun testReadFromJavaProtoBuf() {
+        val javaFoo = pbandk.testpb.java.Test.Foo.getDefaultInstance()
+        val output = ByteArrayOutputStream(100 * 1000)
+
+        // Write protobuf-java Foo to an output stream
+        javaFoo.writeDelimitedTo(output)
+
+        // Read pbandk Foo from the stream
+        val result = output.toByteArray()
+        val input = ByteArrayInputStream(result)
+        val pbFoo = pbandk.testpb.Foo.decodeDelimitedFromStream(input)
+        assertNotNull(pbFoo)
+
+        // Ensure the two instances are equal to each other
+        kotlinJavaRoundtripTest(javaFoo, pbFoo, pbandk.testpb.Foo)
+    }
+
+    // TODO use runtime/src/jvmTest/kotlin/pbandk/TestUtils.kt when available
```

Comment:
I actually just merged another PR that also included this method (sorry, I'd forgotten that that other PR also included this change). If you rebase onto latest `master`, you should be able to use the copy of `kotlinJavaRoundtripTest()` from there.

Reply:
After doing a rebase im getting an error message:

./gradlew clean assemble

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':pbandk-runtime:compileDebugAidl'.
> SDK location not found. Define location with an ANDROID_SDK_ROOT environment variable or by setting the sdk.dir path in your project's local properties file at '/home/snake/dev/github/pbandk/local.properties'.

sorry, kotlin builds are new to me.

Reply:
This is because pbandk depends on the Android toolchain to build the `pbandk-runtime-android` variant. There are instructions at https://developer.android.com/studio#downloads for installing all of Android Studio (if you want) or just the Android CLI tools (the minimum required to compile pbandk). Once you have the Android tools installed, you need to create a `/home/snake/dev/github/pbandk/local.properties` file with an `sdk.dir` property that points at the location where you have the Android SDK tools installed. For example, on my Mac the `local.properties` file contains:

```
sdk.dir=/Users/gary/Library/Android/sdk
```

A couple alternatives if you don't want to install the Android SDK:
1. If you comment out the two `android { }` blocks in `pbandk/runtime/build.gradle.kts` then you can probably run `./gradlew clean assemble` without requiring any Android tools. You might need to comment out the `com.android.library` plugin at the top of the gradle file too.
2. Instead of using the gradle `assemble` task, which will build all supported platform variants, you can use the `jvmJar` or `jsIrJar` tasks to build just the JVM/JS pbandk variants and avoid building the Android variant. There is likewise a `jvmTest` task to only run the unit tests on the JVM variant and avoid the other platforms.